src/inputs/NumberField.tsx

```diff
@@ -71,6 +73,10 @@ export function NumberField(props: NumberFieldProps) {
   // If formatOptions isn't memo'd, a useEffect in useNumberStateField will cause jank,
   // see: https://github.com/adobe/react-spectrum/issues/1893.
   const formatOptions: Intl.NumberFormatOptions | undefined = useMemo(() => {
+    if (props.numberFormatOptions !== undefined) {
```

Comment:
**suggestion (non-blocking):** it would be nice to add & use the `isDefined` helper like we have in internal-frontend.